dig/dig.go

```diff
@@ -120,6 +127,13 @@ func (g *Graph) Resolve(obj interface{}) (err error) {
 	return nil
 }
 
+// MustResolve calls Resolve and panics if an error is encountered
+func (g *Graph) MustResolve(obj interface{}) {
```

Comment:
Well, we want to catch the panic to better decorate it, right? I'd much rather prefer `failed to register *Type1: something something reflect` message rather than a message `something something reflect went boom`.

So I think we won't be able to get away from deferring a recovery anyways, even if we slightly refactor the catch.

To your earlier point, the Try API is going to read weird and be a breaking change to the existing usage.

We probably could go `TryToRegister` and `TryToResolve`, since they read more naturally. I don't have a strong preference, I just want the option to explicit panic if as I developer I know that I can't proceed without a given Register succeeding